features/cellular/framework/API/CellularDevice.h

```diff
@@ -76,36 +76,61 @@ class CellularDevice {
      */
     virtual ~CellularDevice();
 
-    /** Set cellular device power supply on.
+    /** Sets the modem up for powering on
+     *  This is equivalent to plugging in the device, i.e., attaching power and serial port.
+     *  In general, hard_power_on and soft_power_on provides a simple hardware abstraction layer
+     *  on top of the modem drivers written for Mbed OS, which should be used when the modem exists on-board,
+     *  i.e., the modem is NOT a plugged-in component.
```

Comment:
This isn't quite right - they're an abstraction point that can be used for any sort of modem installation. You could make a driver for a custom add-on board with power controls too by deriving from the base driver and overridding these.

Reply:
How about "; they can be overridden in a derived class to perform custom power controls in a particular board configuration."

Reply:
That's excellent, thanks.